src/odemis/gui/cont/acquisition/cryo_z_localization.py

```diff
@@ -248,8 +248,7 @@ def _on_measure_z_done(self, f):
 
             # Update the feature Z pos, and move there
             feature = self._tab_data.main.currentFeature.value
-            pos = feature.pos.value[:2]
-            feature.pos.value = pos + (zpos,)
+            feature.focus_pos.value["z"] = zpos
```

Comment:
Such assignment skips the VA setter. So it skips the checks (but they are none currently), and more importantly doesn't call the subscribers. Typically, you need to always write like:
`feature.focus_pos.value = {"z": zpos}`